constellation-app/constellation
# Handle K-Truss interactive state changes (Feature/issue#62)

# Description of the Change

1. Capture the cases where an open K-Truss panel is closed, If this panel was actively displaying interactive K-Truss visuals, then disable interactive mode on close (ensuring graph is redrawn without K-truss colours/shading).
2. If interactive is toggled off, ensure K-Truss colours are cleared. 

### Alternate Designs

nil

### Why Should This Be In Core?

This addresses an issue described in description by which toggling K-truss interactive flag (or closing K-truss component) was leaving K-Truss visualisations on the graph.

### Benefits

See above, basically, if K-truss interactive is turned off then graph is not cluttered with K-truss information without context.

### Possible Drawbacks

Not aware of any.

### Verification Process

To verify toggling interactive doesn't leave dangling colours:
1. Open a graph with suitable clustering data for display.
2. Select Tools->Cluster->K-Truss.
3. Press the Cluster button. (Interactive is on by default)
4. Check Colour Nested K-Trusses.
5. Turn off interactive by pressing the Interactive button.
Confirm that colours are only shown if interactive is turned on.

To verify closing K-Truss form removes coliuring/shading:
1. Open a graph with suitable clustering data for display.
2. Select Tools->Cluster->K-Truss.
3. Press the Cluster button. (Interactive is on by default)
4. Close the K-Truss component (hitting X in tab header).
Confirm colours and shading are only shown if K-truss component displayed, and interactive is on.

### Applicable Issues

N/A
